engine/runtime/src/main/java/org/enso/interpreter/node/callable/resolver/HostMethodCallNode.java

```diff
@@ -117,6 +117,46 @@ public boolean isInteropLibrary() {
     }
   }
 
+  /**
+   * Given a polyglot (foreign) object, this enum represents a target Enso type that the object
+   * should be converted to before further dispatch.
+   *
+   * <p>For example, a {@code java.lang.String}, or any other polyglot object that {@link
+   * InteropLibrary#isString(Object) is a string}, should be converted to the {@link
+   * org.enso.interpreter.runtime.data.text.Text} builtin Enso type.
+   */
+  public enum PolyglotConversionType {
```

Comment:
- Why do we have `PolyglotConversionType` instead of existing `PolyglotCallType`?
- Well, it is not instead, it is in addition to. Cannot we integrate those two somehow?
- What are the differences preventing such an integration of these two types?